### ISSUE TITLE
Prompt Advisor

### ISSUE THREAD

**Issue by biswapm** [microsoft/Power-CAT-Copilot-Studio-Kit#83]
Prompt advisor to rewrite the prompt or fine tuning the prompt with different prompt technique.

**Comment by patilravikiran** [comment]
This is available in Dec release. Please see [latest release](https://github.com/microsoft/Power-CAT-Copilot-Studio-Kit/releases/tag/CopilotStudioAccelerator-December2024) to see all changes or just download directly via https://aka.ms/DownloadCopilotStudioKit

**Comment by biswapm** [comment]
Prompt advisor setup documentation https://github.com/microsoft/Power-CAT-Copilot-Studio-Kit/blob/main/PROMPT_ADVISOR.md

**Comment by douglasromao** [comment]
Hi team, is there any specific environment requirements to import this solution? I am getting an error when trying to import on my environments.

**Comment by allandecastro** [comment]
> Hi team, is there any specific environment requirements to import this solution? I am getting an error when trying to import on my environments.

Hey @douglasromao 
Can you post the error you encountered? I just installed it on a Dataverse only environment

**Comment by allandecastro** [comment]
@biswapm : I'm struggling to make it work, when I use the custom page and click on Analyze, nothing seems to happen. I've tried using the AI model directly and it works.

From the custom page:

![Image](https://github.com/user-attachments/assets/a7f5f4c8-4aa7-4514-9600-2475f31d2986)


From the model itself:

![Image](https://github.com/user-attachments/assets/ae6d3070-b1d8-4329-97af-3327699b837a)

**Comment by biswapm** [comment]
> [@biswapm](https://github.com/biswapm) : I'm struggling to make it work, when I use the custom page and click on Analyze, nothing seems to happen. I've tried using the AI model directly and it works.
> 
> From the custom page:
> 
> ![Image](https://github.com/user-attachments/assets/a7f5f4c8-4aa7-4514-9600-2475f31d2986)
> 
> From the model itself:
> 
> ![Image](https://github.com/user-attachments/assets/ae6d3070-b1d8-4329-97af-3327699b837a)

@allandecastro, does it stay in "waiting" to get response or you're getting any error? For the 1st time load we have observed API is taking some time to respond back.

**Comment by biswapm** [comment]
> Hi team, is there any specific environment requirements to import this solution? I am getting an error when trying to import on my environments.

@douglasromao - Please share the screenshot, however you can find Pre-requisites here:  https://github.com/microsoft/Power-CAT-Copilot-Studio-Kit/blob/main/PROMPT_ADVISOR.md#prerequisites.

cc @psimolin

**Comment by allandecastro** [comment]
> [@allandecastro](https://github.com/allandecastro), does it stay in "waiting" to get response or you're getting any error? For the 1st time load we have observed API is taking some time to respond back.

@biswapm: Thanks for your answer! It stays in "Waiting for prompt to evaluate.". I don't even know if the button "Analyze" trigger something or not at this stage. What should the UpdateContext trigger?

**Comment by patilravikiran** [comment]
Hi @allandecastro , 
It seems Power Apps Component Framework feature is not enabled for your environment. Please enable same using [Enable the Power Apps component framework feature](https://learn.microsoft.com/en-us/power-apps/developer/component-framework/component-framework-for-canvas-apps#enable-the-power-apps-component-framework-feature) and try reinstalling solution again. 

Let us know if you are still facing this issue.

**Comment by allandecastro** [comment]
@patilravikiran : You're correct, it works like a charm :) Think it will be great to add it as a prerequisite here (sorry can't do a PR right now) : https://github.com/microsoft/Power-CAT-Copilot-Studio-Kit/blob/main/PROMPT_ADVISOR.md#prerequisites

**Comment by biswapm** [comment]
> [@patilravikiran](https://github.com/patilravikiran) : You're correct, it works like a charm :) Think it will be great to add it as a prerequisite here (sorry can't do a PR right now) : https://github.com/microsoft/Power-CAT-Copilot-Studio-Kit/blob/main/PROMPT_ADVISOR.md#prerequisites

@allandecastro Thank you for the great feedback and collaboration!! We have just updated the doc.

**Comment by patilravikiran** [comment]
Closing this issue as changes are already available within latest release.
